Assets/__Scripts/MapEditor/Grid/Collections/EventsContainer.cs

```diff
@@ -18,13 +18,24 @@ public bool RingPropagationEditing
         set
         {
             ringPropagationEditing = value;
-            ringPropagationLabels.SetActive(value);
-            eventGridLabels.SetActive(!value);
+            PlatformDescriptor pD = FindObjectOfType<PlatformDescriptor>();
+            eventGridLabels.transform.parent.GetComponent<CreateEventTypeLabels>().UpdateLabels(value, value ? (pD.BigRingManager?.rings.Length ?? 15)+1 : 16);
```

Comment:
Make a `CreateEventTypeLabels` reference and assign it in the Unity Editor instead of grabbing the component manually:

```cs
[SerializeField] private CreateEventTypeLabels labels;
```